From: Prarit Bhargava <prarit@redhat.com>
Subject: Re: [RHEL5.1 PATCH] BZ 226849 : remove __init from efi_get_time
Date: Mon, 19 Feb 2007 13:31:21 -0500
Bugzilla: 226849
Message-Id: <45D9ECF9.7050102@redhat.com>
Changelog: [x86] remove __init from efi_get_time

Remove __init from efi_get_time.

Resolves BZ 226849.

Successfully tested by me.

--- linux-2.6.18.x86_64.orig/arch/i386/kernel/efi.c	2006-09-19 23:42:06.000000000 -0400
+++ linux-2.6.18.x86_64//arch/i386/kernel/efi.c	2007-02-19 13:26:56.000000000 -0500
@@ -198,7 +198,7 @@ inline int efi_set_rtc_mmss(unsigned lon
  * services have been remapped, therefore, we'll need to call in physical
  * mode.  Note, this call isn't used later, so mark it __init.
  */
-inline unsigned long __init efi_get_time(void)
+inline unsigned long efi_get_time(void)
 {
 	efi_status_t status;
 	efi_time_t eft;
--- linux-2.6.18.x86_64.orig/include/linux/efi.h	2006-09-19 23:42:06.000000000 -0400
+++ linux-2.6.18.x86_64/include/linux/efi.h	2007-02-19 12:02:54.000000000 -0500
@@ -300,8 +300,8 @@ extern int efi_mem_attribute_range (unsi
 extern int __init efi_uart_console_only (void);
 extern void efi_initialize_iomem_resources(struct resource *code_resource,
 					struct resource *data_resource);
-extern unsigned long __init efi_get_time(void);
-extern int __init efi_set_rtc_mmss(unsigned long nowtime);
+extern unsigned long efi_get_time(void);
+extern int efi_set_rtc_mmss(unsigned long nowtime);
 extern struct efi_memory_map memmap;
 
 /**

